### PR TITLE
capture a single ref alias

### PR DESCRIPTION
**Pull request by davidrichardson**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/repository/model/templates/SingleRefCapture.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Override
+    public List<String> expectedColumnHeaders() {
+        return Collections.emptyList();
+    }
+
+    private void addRef(String value, JSONObject document){
+        JSONObject  ref = new JSONObject();
+        ref.put("alias",value);
+        document.put(refKey,ref);
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/repository/model/templates/SingleRefCapture.java:47]: Are we sure that the 'refKey' variable can't be null?

File: src/main/java/uk/ac/ebi/subs/repository/model/templates/SingleRefCapture.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public int map(int position, List<Capture> captures, List<String> headers) {
+
+        this.setCaptureInList(position,captures,headers.get(position));
+
+        return ++position;
+    }
+
+    @Override
+    public List<String> expectedColumnHeaders() {
+        return Collections.emptyList();
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/repository/model/templates/SingleRefCapture.java]: Is this always an empty list?

**Review by ke4 — CHANGES_REQUESTED** [review]
I added some comments

File: src/main/java/uk/ac/ebi/subs/repository/model/templates/SingleRefCapture.java (hunk 2 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/repository/model/templates/SingleRefCapture.java]: Yes. Some of 'capture' classes take multiple columns, after primary the named column (e.g. attributes have the main value column, + optional units and terms columns.

I'll move this function to be the interface default, as there's no point replicating it all over the place

File: src/main/java/uk/ac/ebi/subs/repository/model/templates/SingleRefCapture.java (hunk 1 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/repository/model/templates/SingleRefCapture.java:47]: I'll add a constraint to ensure it isn't null

**Review by ke4 — APPROVED** [review]
Looks fine with me 👍

**Pull request merged**